web/src/components/taxonomies/taxonomies-tree.tsx

```diff
@@ -53,23 +55,27 @@ export const TaxonomiesTree: FC<TaxonomiesTreeProps> = ({
         [expandNode]
     );
 
-    const handleSelect = useCallback((selectedKeys, info) => {
-        setSelectedKeys(selectedKeys);
-        if (selectedAnnotation?.labels) {
-            const labelToChangeIdx: number = selectedAnnotation?.labels.findIndex(
-                (label: AnnotationLabel) => label.annotationId === selectedAnnotation.id
-            );
+    const handleSelect = useCallback(
+        (handleSelectedKeys, info) => {
+            selectedKeys = handleSelectedKeys;
```

Comment:
hmm, it's look like antipattern for react apps, instead it would be better to change selectedKeys to state

Reply:
but these changes fixed problems with multiple re-renders. Could you send me a link for this antipattern, pls? I want to read more about this

Reply:
it helps with state in this case. Another solution requires a deep refactoring of the project